pkg/vadmin/restore_points_query_vc.go

```diff
@@ -17,9 +17,62 @@ package vadmin
 
 import (
 	"context"
+	"fmt"
+
+	vops "github.com/vertica/vcluster/vclusterops"
+	"github.com/vertica/vcluster/vclusterops/vstruct"
+	"github.com/vertica/vertica-kubernetes/pkg/net"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/restorepoints"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-func (v *VClusterOps) ListRestorePoints(_ context.Context) error {
-	v.Log.Info("start restore points query")
-	return nil
+// RestartNode will restart a subset of nodes. Use this when vertica has not
+// lost cluster quorum. The IP given for each vnode may not match the current IP
+// in the vertica catalogs.
+func (v *VClusterOps) ListRestorePoints(ctx context.Context, opts ...restorepoints.Option) (ctrl.Result, error) {
+	v.setupForAPICall("RestorePoints")
+	defer v.tearDownForAPICall()
+	v.Log.Info("Starting vcluster RestorePoints")
+
+	certs, err := v.retrieveNMACerts(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	s := restorepoints.Parms{}
+	s.Make(opts...)
+
+	vcOpts := v.genRestorePointsOptions(&s, certs)
+	_, err = v.VShowRestorePoints(vcOpts)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to show restore points: %w", err)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (v *VClusterOps) genRestorePointsOptions(s *restorepoints.Parms, certs *HTTPSCerts) *vops.VShowRestorePointsOptions {
+	opts := vops.VShowRestorePointsFactory()
+
+	// required options
+	opts.DBName = &v.VDB.Spec.DBName
+	opts.CommunalStorageLocation = &s.CommunalPath
+
+	*opts.HonorUserInput = true
+	opts.RawHosts = []string{s.InitiatorIP}
+	opts.Hosts = s.Hosts
+	v.Log.Info("Setup restore point options", "rawhosts", opts.RawHosts)
+	v.Log.Info("Setup restore point options", "hosts", opts.Hosts)
+
+	opts.Ipv6 = vstruct.MakeNullableBool(net.IsIPv6(s.InitiatorIP))
+	opts.ConfigurationParameters = s.ConfigurationParams
+
+	// auth options
+	opts.Key = certs.Key
+	opts.Cert = certs.Cert
+	opts.CaCert = certs.CaCert
+	*opts.UserName = v.VDB.GetVerticaUser()
```

Comment:
Why do we need the user/password to get then restore points? I thought it just talked to the NMA.